test/integration/agent/gobgp_test.go

```diff
@@ -0,0 +1,436 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/ptr"
+
+	"antrea.io/antrea/pkg/agent/bgp"
+	"antrea.io/antrea/pkg/agent/bgp/gobgp"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+)
+
+func TestGoBGPLifecycle(t *testing.T) {
+	asn1 := int32(61179)
+	asn2 := int32(62179)
+	asn3 := int32(63179)
+	routerID1 := "192.168.1.1"
+	routerID2 := "192.168.1.2"
+	routerID3 := "192.168.1.3"
+	listenPort1 := int32(1179)
+	listenPort2 := int32(2179)
+	listenPort3 := int32(3179)
+	server1GlobalConfig := &bgp.GlobalConfig{
+		ASN:        uint32(asn1),
+		RouterID:   routerID1,
+		ListenPort: listenPort1,
+	}
+	server2GlobalConfig := &bgp.GlobalConfig{
+		ASN:        uint32(asn2),
+		RouterID:   routerID2,
+		ListenPort: listenPort2,
+	}
+	server3GlobalConfig := &bgp.GlobalConfig{
+		ASN:        uint32(asn3),
+		RouterID:   routerID3,
+		ListenPort: listenPort3,
+	}
+
+	server1 := gobgp.NewGoBGPServer(server1GlobalConfig)
+	server2 := gobgp.NewGoBGPServer(server2GlobalConfig)
+	server3 := gobgp.NewGoBGPServer(server3GlobalConfig)
+
+	ctx := context.TODO()
+	t.Log("Starting all BGP servers")
+	require.NoError(t, server1.Start(ctx))
+	require.NoError(t, server2.Start(ctx))
+	require.NoError(t, server3.Start(ctx))
+	t.Log("Started all BGP servers")
+
+	ipv4Server1Config := bgp.PeerConfig{
+		BGPPeer: &v1alpha1.BGPPeer{
+			Address:                    "127.0.0.1",
+			Port:                       ptr.To(int32(1179)),
```

Comment:
nit: I personally prefer `ptr.To[int32](...)` in this case, but I have seen both in the K8s codebase

Reply:
`ptr.To[int32](...)` looks cleaner and neater.